### PR TITLE
Bugfix: MusicSlider with GameManager

### PR DESCRIPTION
**Pull request by AHinojal**
Now, we can control the audio between the scenes with a Game Manager (that we can use also for the points or any variable that we need to save between the scenes during the execution of the game)

To install the GameManager, you have to drag&drop the prefab `GameManager.prefab` to the scene and inside it in the inspector, linked it. Also we have to link it in the slider of the audio (Panels>CanvasSettings>PanelAudio>PanelAudio>Slider in StartScene p.e) in VolumeControl's script.

We have to do that in all the scenes.

TESTING SCENES WITH GAME MANAGER: 
[Scenes.zip](https://github.com/PenteractStudios/Penteract/files/6707878/Scenes.zip)

### CODE REVIEW

**Comment by jcarlos0305** [comment]
Task linked: [CU-kxdhwu MusicSlider bug](https://app.clickup.com/t/kxdhwu)

**Review by lowysole — APPROVED** [review]

**Review by isaacfdez — APPROVED** [review]

File: Penteract/Assets/Scripts/VolumeControl.cpp [hunk 1]
```diff
@@ -5,24 +5,38 @@
...
 	if (sliderGO && gameCamera) {
 		ComponentSlider* comSlider = sliderGO->GetComponent<ComponentSlider>();
 		ComponentAudioListener* comAudioLis = gameCamera->GetComponent<ComponentAudioListener>();
 		if (comSlider && comAudioLis) {
-			float volume = comSlider->GetNormalizedValue();
+			float volume = 0.f; 
+
+			// To control volume between scenes in GameManager
+			if (gameManager) {
+				gameManagerController = GET_SCRIPT(gameManager, GameManager);
```
Comment (Mefiso) [inline Penteract/Assets/Scripts/VolumeControl.cpp:31]: You are not using gameManagerController, so why are getting it?

File: Penteract/Assets/Scripts/VolumeControl.cpp [hunk 2]
```diff
@@ -37,7 +51,14 @@ void VolumeControl::OnValueChanged() {
 	ComponentSlider* comSlider = sliderGO->GetComponent<ComponentSlider>();
 	ComponentAudioListener* comAudioLis = gameCamera->GetComponent<ComponentAudioListener>();
 	if (!comSlider || !comAudioLis) return;
-
+	
 	float volume = comSlider->GetNormalizedValue();
+	
+	// To control volume between scenes in GameManager
+	if (gameManager) {
+		gameManagerController = GET_SCRIPT(gameManager, GameManager);
```
Comment (Mefiso) [inline Penteract/Assets/Scripts/VolumeControl.cpp:59]: Same

**Review by Mefiso — COMMENTED** [review]
Maybe I'm missing something, but, wouldn't it be simpler the following? GameManager does not inherit from GameSingleton; Put the "static GameSingleton* instance" and "static GameSingleton* GetInstance()" inside GameManager directly; remove duplicated parameter inside GameManager. 
Because I think that having a GameManager, that is a GameSingleton, but only cares about a reference to another GameSingleton (violating the singleton pattern), which is the true singleton, and that has a repeated parameter that copies in every update from that reference of the singleton... is a bit confusing...

Instead, I'm suggesting that the singleton is just the bunch of parameters (and the getters if  you want), and then the GameManager has that static reference that makes the GameSingleton class a singleton. But then again, maybe I'm missing something and that wouldn't work.

File: Penteract/Assets/Scripts/GameManager.h [hunk 3]
```diff
@@ -0,0 +1,21 @@
...
+public:
+
+	void Start() override;
+	void Update() override;
+
+public:
+	UID gameManagerUID = 0;
+
+	// Put the same values that in GameSingleton.h
+	float volumeScene = 0.5f;
```
Comment (Mefiso) [inline Penteract/Assets/Scripts/GameManager.h:20]: Are you sure you need this and then set it in every update?

**Pull request closed without merging**